pywbem/_mof_compiler.py

```diff
@@ -2672,7 +2674,15 @@ def compile_string(self, mof, ns, filename=None):
         except AttributeError:
             oldmof = None
         self.parser.mof = mof
+
+        # Save default_namepace so it can be restored after the parse.
+        # Note that there may be nested calls to compile_string
+        try:
+            oldns = self.parser.handle.default_namespace
+        except AttributeError:
+            oldns = None
         self.parser.handle.default_namespace = ns
```

Comment:
Besides the previous comment, can we not avoid the modification of the `default_namespace` altogether? We could set the target namespace as a new attribute on `self.parser`, like so:
```
self.parser.target_namespace = ns
```
and every time the parser invokes an operation on its `self.handle`, pass the `self.target_namespace` to the operation explicitly.

Reply:
Refactored the MOF compiler to use a target_namespace and not have any predefined use of the default namespace, and refactored MOFWBEMConnection and MOCKMOFWBEMConnection to support a namespace  parameter on the GetClass, etc calls so that compiler can pas the defined namespace back to these methods.

Reply:
Ok, you were right, it was more than just what I suggested above. Thanks for this change!